integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/CreatePreAggregateTableCommand.scala

```diff
@@ -34,105 +32,74 @@ import org.apache.carbondata.core.util.CarbonUtil
-      if (createDSTable) {
-        try {
-          val fields = new Array[Field](cm.dimCols.size + cm.msrCols.size)
-          cm.dimCols.foreach(f => fields(f.schemaOrdinal) = f)
-          cm.msrCols.foreach(f => fields(f.schemaOrdinal) = f)
-          sparkSession.sql(
-            s"""CREATE TABLE $dbName.$tbName
-               |(${ fields.map(f => f.rawSchema).mkString(",") })
-               |USING org.apache.spark.sql.CarbonSource""".stripMargin +
-            s""" OPTIONS (tableName "$tbName", dbName "$dbName", tablePath """.stripMargin +
-            s""""$tablePath"$carbonSchemaString) """)
-          // child schema object which will be updated on parent table about the
-          val childSchema = tableInfo.getFactTable
-            .buildChildSchema("", tableInfo.getDatabaseName, queryString, "AGGREGATION")
-          // upadting the parent table about child table
-          PreAggregateUtil.updateMainTable(parentDbName, parentTableName, childSchema, sparkSession)
-          val loadAvailable = PreAggregateUtil
-            .checkMainTableLoad(parentTable)
-          if (loadAvailable) {
-            sparkSession.sql(s"insert into ${ cm.databaseName }.${ cm.tableName } $queryString")
-          }
-        } catch {
-          case e: Exception =>
-            val identifier: TableIdentifier = TableIdentifier(tbName, Some(dbName))
-            // call the drop table to delete the created table.
-            CarbonEnv.getInstance(sparkSession).carbonMetastore
-              .dropTable(tablePath, identifier)(sparkSession)
-            LOGGER.audit(s"Table creation with Database name [$dbName] " +
-                         s"and Table name [$tbName] failed")
-            throw e
-        }
+    tableModel.parentTable = Some(parentTable)
+    tableModel.dataMapRelation = Some(fieldRelationMap)
+    CarbonCreateTableCommand(tableModel).run(sparkSession)
+    try {
+      val relation = CarbonEnv.getInstance(sparkSession).carbonMetastore.
+        lookupRelation(tableIdentifier)(sparkSession).asInstanceOf[CarbonRelation]
+      val tableInfo = relation.tableMeta.carbonTable.getTableInfo
+      // child schema object which will be updated on parent table about the
+      val childSchema = tableInfo.getFactTable
+        .buildChildSchema(dataMapName, "", tableInfo.getDatabaseName, queryString, "AGGREGATION")
+      dmproperties.foreach(f => childSchema.getProperties.put(f._1, f._2))
+      // updating the parent table about child table
+      PreAggregateUtil.updateMainTable(parentDbName, parentTableName, childSchema, sparkSession)
+      val loadAvailable = PreAggregateUtil.checkMainTableLoad(parentTable)
+      if (loadAvailable) {
+        sparkSession
+          .sql(s"insert into ${ tableModel.databaseName }.${ tableModel.tableName } $queryString")
       }
+    } catch {
+      case e: Exception =>
+        sparkSession.
+          sql(s"""DROP TABLE IF EXISTS ${ tableModel.databaseName }.${ tableModel.tableName }""")
```

Comment:
move to previous line, move parameter to next line

Reply:
ok